{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload\n",
    "%autoreload 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pickle\n",
    "from glob import glob\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "\n",
    "import settings as conf"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load gene mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(os.path.join(conf.GENES_METADATA_DIR, 'genes_mapping_simplified-0.pkl'), 'rb') as f:\n",
    "    genes_mapping_0 = pickle.load(f)\n",
    "\n",
    "with open(os.path.join(conf.GENES_METADATA_DIR, 'genes_mapping_simplified-1.pkl'), 'rb') as f:\n",
    "    genes_mapping_1 = pickle.load(f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load S-MultiXcan results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/miltondp/projects/labs/hakyimlab/phenomexcan/base/gene_assoc/smultixcan-mashr-pvalues.pkl'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# s-predixcan\n",
    "smultixcan_genes_associations_filename = os.path.join(conf.GENE_ASSOC_DIR, f'smultixcan-mashr-pvalues.pkl')\n",
    "display(smultixcan_genes_associations_filename)\n",
    "\n",
    "smultixcan_genes_associations = pd.read_pickle(smultixcan_genes_associations_filename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(22515, 4091)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "smultixcan_genes_associations.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>20096_1-Size_of_red_wine_glass_drunk_small_125ml</th>\n",
       "      <th>2345-Ever_had_bowel_cancer_screening</th>\n",
       "      <th>N49-Diagnoses_main_ICD10_N49_Inflammatory_disorders_of_male_genital_organs_not_elsewhere_classified</th>\n",
       "      <th>100011_raw-Iron</th>\n",
       "      <th>5221-Index_of_best_refractometry_result_right</th>\n",
       "      <th>20003_1141150624-Treatmentmedication_code_zomig_25mg_tablet</th>\n",
       "      <th>S69-Diagnoses_main_ICD10_S69_Other_and_unspecified_injuries_of_wrist_and_hand</th>\n",
       "      <th>20024_1136-Job_code_deduced_Information_and_communication_technology_managers</th>\n",
       "      <th>20002_1385-Noncancer_illness_code_selfreported_allergy_or_anaphylactic_reaction_to_food</th>\n",
       "      <th>G6_SLEEPAPNO-Sleep_apnoea</th>\n",
       "      <th>...</th>\n",
       "      <th>Astle_et_al_2016_Sum_basophil_neutrophil_counts</th>\n",
       "      <th>RA_OKADA_TRANS_ETHNIC</th>\n",
       "      <th>pgc.scz2</th>\n",
       "      <th>PGC_ADHD_EUR_2017</th>\n",
       "      <th>MAGIC_FastingGlucose</th>\n",
       "      <th>Astle_et_al_2016_Red_blood_cell_count</th>\n",
       "      <th>SSGAC_Depressive_Symptoms</th>\n",
       "      <th>BCAC_ER_positive_BreastCancer_EUR</th>\n",
       "      <th>IBD.EUR.Inflammatory_Bowel_Disease</th>\n",
       "      <th>Astle_et_al_2016_High_light_scatter_reticulocyte_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000000419</th>\n",
       "      <td>0.865429</td>\n",
       "      <td>0.918314</td>\n",
       "      <td>0.810683</td>\n",
       "      <td>0.374671</td>\n",
       "      <td>0.189032</td>\n",
       "      <td>0.140981</td>\n",
       "      <td>0.467741</td>\n",
       "      <td>0.129427</td>\n",
       "      <td>0.193680</td>\n",
       "      <td>0.285479</td>\n",
       "      <td>...</td>\n",
       "      <td>0.416210</td>\n",
       "      <td>0.782554</td>\n",
       "      <td>0.609467</td>\n",
       "      <td>0.980281</td>\n",
       "      <td>0.666504</td>\n",
       "      <td>0.409761</td>\n",
       "      <td>0.713310</td>\n",
       "      <td>0.168319</td>\n",
       "      <td>0.460244</td>\n",
       "      <td>0.765506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000457</th>\n",
       "      <td>0.174192</td>\n",
       "      <td>0.064765</td>\n",
       "      <td>0.889194</td>\n",
       "      <td>0.896938</td>\n",
       "      <td>0.448596</td>\n",
       "      <td>0.269602</td>\n",
       "      <td>0.540261</td>\n",
       "      <td>0.068405</td>\n",
       "      <td>0.041813</td>\n",
       "      <td>0.313427</td>\n",
       "      <td>...</td>\n",
       "      <td>0.149360</td>\n",
       "      <td>0.512603</td>\n",
       "      <td>0.010907</td>\n",
       "      <td>0.228982</td>\n",
       "      <td>0.607081</td>\n",
       "      <td>0.812484</td>\n",
       "      <td>0.678749</td>\n",
       "      <td>0.918971</td>\n",
       "      <td>0.311187</td>\n",
       "      <td>0.344574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000460</th>\n",
       "      <td>0.879969</td>\n",
       "      <td>0.240715</td>\n",
       "      <td>0.238228</td>\n",
       "      <td>0.567555</td>\n",
       "      <td>0.921320</td>\n",
       "      <td>0.825036</td>\n",
       "      <td>0.782230</td>\n",
       "      <td>0.644525</td>\n",
       "      <td>0.392273</td>\n",
       "      <td>0.840014</td>\n",
       "      <td>...</td>\n",
       "      <td>0.503520</td>\n",
       "      <td>0.764147</td>\n",
       "      <td>0.587969</td>\n",
       "      <td>0.301460</td>\n",
       "      <td>0.629621</td>\n",
       "      <td>0.486664</td>\n",
       "      <td>0.736509</td>\n",
       "      <td>0.933600</td>\n",
       "      <td>0.000477</td>\n",
       "      <td>0.321223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000938</th>\n",
       "      <td>0.192670</td>\n",
       "      <td>0.400054</td>\n",
       "      <td>0.114353</td>\n",
       "      <td>0.470700</td>\n",
       "      <td>0.889202</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>0.899764</td>\n",
       "      <td>0.212352</td>\n",
       "      <td>0.829671</td>\n",
       "      <td>0.372348</td>\n",
       "      <td>...</td>\n",
       "      <td>0.899212</td>\n",
       "      <td>0.961678</td>\n",
       "      <td>0.059247</td>\n",
       "      <td>0.588855</td>\n",
       "      <td>0.898525</td>\n",
       "      <td>0.135045</td>\n",
       "      <td>0.954998</td>\n",
       "      <td>0.088220</td>\n",
       "      <td>0.176497</td>\n",
       "      <td>0.304281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000000971</th>\n",
       "      <td>0.180632</td>\n",
       "      <td>0.793060</td>\n",
       "      <td>0.490585</td>\n",
       "      <td>0.088752</td>\n",
       "      <td>0.744531</td>\n",
       "      <td>0.949639</td>\n",
       "      <td>0.253817</td>\n",
       "      <td>0.377408</td>\n",
       "      <td>0.971655</td>\n",
       "      <td>0.070266</td>\n",
       "      <td>...</td>\n",
       "      <td>0.390618</td>\n",
       "      <td>0.093824</td>\n",
       "      <td>0.020391</td>\n",
       "      <td>0.109883</td>\n",
       "      <td>0.870551</td>\n",
       "      <td>0.995450</td>\n",
       "      <td>0.002660</td>\n",
       "      <td>0.421588</td>\n",
       "      <td>0.656851</td>\n",
       "      <td>0.868416</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 4091 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 20096_1-Size_of_red_wine_glass_drunk_small_125ml  \\\n",
       "gene_name                                                           \n",
       "ENSG00000000419                                          0.865429   \n",
       "ENSG00000000457                                          0.174192   \n",
       "ENSG00000000460                                          0.879969   \n",
       "ENSG00000000938                                          0.192670   \n",
       "ENSG00000000971                                          0.180632   \n",
       "\n",
       "                 2345-Ever_had_bowel_cancer_screening  \\\n",
       "gene_name                                               \n",
       "ENSG00000000419                              0.918314   \n",
       "ENSG00000000457                              0.064765   \n",
       "ENSG00000000460                              0.240715   \n",
       "ENSG00000000938                              0.400054   \n",
       "ENSG00000000971                              0.793060   \n",
       "\n",
       "                 N49-Diagnoses_main_ICD10_N49_Inflammatory_disorders_of_male_genital_organs_not_elsewhere_classified  \\\n",
       "gene_name                                                                                                              \n",
       "ENSG00000000419                                           0.810683                                                     \n",
       "ENSG00000000457                                           0.889194                                                     \n",
       "ENSG00000000460                                           0.238228                                                     \n",
       "ENSG00000000938                                           0.114353                                                     \n",
       "ENSG00000000971                                           0.490585                                                     \n",
       "\n",
       "                 100011_raw-Iron  \\\n",
       "gene_name                          \n",
       "ENSG00000000419         0.374671   \n",
       "ENSG00000000457         0.896938   \n",
       "ENSG00000000460         0.567555   \n",
       "ENSG00000000938         0.470700   \n",
       "ENSG00000000971         0.088752   \n",
       "\n",
       "                 5221-Index_of_best_refractometry_result_right  \\\n",
       "gene_name                                                        \n",
       "ENSG00000000419                                       0.189032   \n",
       "ENSG00000000457                                       0.448596   \n",
       "ENSG00000000460                                       0.921320   \n",
       "ENSG00000000938                                       0.889202   \n",
       "ENSG00000000971                                       0.744531   \n",
       "\n",
       "                 20003_1141150624-Treatmentmedication_code_zomig_25mg_tablet  \\\n",
       "gene_name                                                                      \n",
       "ENSG00000000419                                           0.140981             \n",
       "ENSG00000000457                                           0.269602             \n",
       "ENSG00000000460                                           0.825036             \n",
       "ENSG00000000938                                           0.000011             \n",
       "ENSG00000000971                                           0.949639             \n",
       "\n",
       "                 S69-Diagnoses_main_ICD10_S69_Other_and_unspecified_injuries_of_wrist_and_hand  \\\n",
       "gene_name                                                                                        \n",
       "ENSG00000000419                                           0.467741                               \n",
       "ENSG00000000457                                           0.540261                               \n",
       "ENSG00000000460                                           0.782230                               \n",
       "ENSG00000000938                                           0.899764                               \n",
       "ENSG00000000971                                           0.253817                               \n",
       "\n",
       "                 20024_1136-Job_code_deduced_Information_and_communication_technology_managers  \\\n",
       "gene_name                                                                                        \n",
       "ENSG00000000419                                           0.129427                               \n",
       "ENSG00000000457                                           0.068405                               \n",
       "ENSG00000000460                                           0.644525                               \n",
       "ENSG00000000938                                           0.212352                               \n",
       "ENSG00000000971                                           0.377408                               \n",
       "\n",
       "                 20002_1385-Noncancer_illness_code_selfreported_allergy_or_anaphylactic_reaction_to_food  \\\n",
       "gene_name                                                                                                  \n",
       "ENSG00000000419                                           0.193680                                         \n",
       "ENSG00000000457                                           0.041813                                         \n",
       "ENSG00000000460                                           0.392273                                         \n",
       "ENSG00000000938                                           0.829671                                         \n",
       "ENSG00000000971                                           0.971655                                         \n",
       "\n",
       "                 G6_SLEEPAPNO-Sleep_apnoea  ...  \\\n",
       "gene_name                                   ...   \n",
       "ENSG00000000419                   0.285479  ...   \n",
       "ENSG00000000457                   0.313427  ...   \n",
       "ENSG00000000460                   0.840014  ...   \n",
       "ENSG00000000938                   0.372348  ...   \n",
       "ENSG00000000971                   0.070266  ...   \n",
       "\n",
       "                 Astle_et_al_2016_Sum_basophil_neutrophil_counts  \\\n",
       "gene_name                                                          \n",
       "ENSG00000000419                                         0.416210   \n",
       "ENSG00000000457                                         0.149360   \n",
       "ENSG00000000460                                         0.503520   \n",
       "ENSG00000000938                                         0.899212   \n",
       "ENSG00000000971                                         0.390618   \n",
       "\n",
       "                 RA_OKADA_TRANS_ETHNIC  pgc.scz2  PGC_ADHD_EUR_2017  \\\n",
       "gene_name                                                             \n",
       "ENSG00000000419               0.782554  0.609467           0.980281   \n",
       "ENSG00000000457               0.512603  0.010907           0.228982   \n",
       "ENSG00000000460               0.764147  0.587969           0.301460   \n",
       "ENSG00000000938               0.961678  0.059247           0.588855   \n",
       "ENSG00000000971               0.093824  0.020391           0.109883   \n",
       "\n",
       "                 MAGIC_FastingGlucose  Astle_et_al_2016_Red_blood_cell_count  \\\n",
       "gene_name                                                                      \n",
       "ENSG00000000419              0.666504                               0.409761   \n",
       "ENSG00000000457              0.607081                               0.812484   \n",
       "ENSG00000000460              0.629621                               0.486664   \n",
       "ENSG00000000938              0.898525                               0.135045   \n",
       "ENSG00000000971              0.870551                               0.995450   \n",
       "\n",
       "                 SSGAC_Depressive_Symptoms  BCAC_ER_positive_BreastCancer_EUR  \\\n",
       "gene_name                                                                       \n",
       "ENSG00000000419                   0.713310                           0.168319   \n",
       "ENSG00000000457                   0.678749                           0.918971   \n",
       "ENSG00000000460                   0.736509                           0.933600   \n",
       "ENSG00000000938                   0.954998                           0.088220   \n",
       "ENSG00000000971                   0.002660                           0.421588   \n",
       "\n",
       "                 IBD.EUR.Inflammatory_Bowel_Disease  \\\n",
       "gene_name                                             \n",
       "ENSG00000000419                            0.460244   \n",
       "ENSG00000000457                            0.311187   \n",
       "ENSG00000000460                            0.000477   \n",
       "ENSG00000000938                            0.176497   \n",
       "ENSG00000000971                            0.656851   \n",
       "\n",
       "                 Astle_et_al_2016_High_light_scatter_reticulocyte_count  \n",
       "gene_name                                                                \n",
       "ENSG00000000419                                           0.765506       \n",
       "ENSG00000000457                                           0.344574       \n",
       "ENSG00000000460                                           0.321223       \n",
       "ENSG00000000938                                           0.304281       \n",
       "ENSG00000000971                                           0.868416       \n",
       "\n",
       "[5 rows x 4091 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "smultixcan_genes_associations.head(5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Create UKB data with traits"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "ukb_data = pd.DataFrame(index=smultixcan_genes_associations.columns).reset_index().rename(columns={'index': 'ukb_fullcode'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "ukb_data = ukb_data.assign(ukb_code=ukb_data['ukb_fullcode'].str.split('-').str[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ukb_fullcode</th>\n",
       "      <th>ukb_code</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20096_1-Size_of_red_wine_glass_drunk_small_125ml</td>\n",
       "      <td>20096_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2345-Ever_had_bowel_cancer_screening</td>\n",
       "      <td>2345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>N49-Diagnoses_main_ICD10_N49_Inflammatory_diso...</td>\n",
       "      <td>N49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>100011_raw-Iron</td>\n",
       "      <td>100011_raw</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5221-Index_of_best_refractometry_result_right</td>\n",
       "      <td>5221</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        ukb_fullcode    ukb_code\n",
       "0   20096_1-Size_of_red_wine_glass_drunk_small_125ml     20096_1\n",
       "1               2345-Ever_had_bowel_cancer_screening        2345\n",
       "2  N49-Diagnoses_main_ICD10_N49_Inflammatory_diso...         N49\n",
       "3                                    100011_raw-Iron  100011_raw\n",
       "4      5221-Index_of_best_refractometry_result_right        5221"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_data.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Take a look at the UKB-EFO mappings file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Note:** `UK_Biobank_master_file.tsv` was downloaded from https://github.com/EBISPOT/EFO-UKB-mappings"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "ukb_map = pd.read_csv(conf.OMIM_SILVER_STANDARD_UKB_EFO_MAP_FILE, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "ukb_map = ukb_map.dropna(subset=['ICD10_CODE/SELF_REPORTED_TRAIT_FIELD_CODE'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1564, 7)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ZOOMA QUERY</th>\n",
       "      <th>MAPPED_TERM_LABEL</th>\n",
       "      <th>MAPPED_TERM_URI</th>\n",
       "      <th>MAPPING_TYPE</th>\n",
       "      <th>ICD10_CODE/SELF_REPORTED_TRAIT_FIELD_CODE</th>\n",
       "      <th>COMMENTS/TICKET</th>\n",
       "      <th>AI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Vascular disorders of intestine</td>\n",
       "      <td>vascular disease</td>\n",
       "      <td>EFO_0004264, EFO_0009431</td>\n",
       "      <td>Broad</td>\n",
       "      <td>K55</td>\n",
       "      <td>DONE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gonarthrosis</td>\n",
       "      <td>osteoarthritis || knee</td>\n",
       "      <td>EFO_0004616</td>\n",
       "      <td>Broad</td>\n",
       "      <td>M17</td>\n",
       "      <td>DONE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Psoriatic and enteropathic arthropathies</td>\n",
       "      <td>psoriatic arthritis</td>\n",
       "      <td>EFO_0003778</td>\n",
       "      <td>? Broad</td>\n",
       "      <td>M07</td>\n",
       "      <td>DONE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Pain associated with micturition</td>\n",
       "      <td>dysuria</td>\n",
       "      <td>EFO_0003901</td>\n",
       "      <td>? Broad</td>\n",
       "      <td>R30</td>\n",
       "      <td>DONE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Other mood</td>\n",
       "      <td>mood disorder</td>\n",
       "      <td>EFO_0004247</td>\n",
       "      <td>? Broad</td>\n",
       "      <td>F38</td>\n",
       "      <td>DONE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                ZOOMA QUERY       MAPPED_TERM_LABEL  \\\n",
       "0           Vascular disorders of intestine        vascular disease   \n",
       "1                              Gonarthrosis  osteoarthritis || knee   \n",
       "2  Psoriatic and enteropathic arthropathies     psoriatic arthritis   \n",
       "3          Pain associated with micturition                 dysuria   \n",
       "4                                Other mood           mood disorder   \n",
       "\n",
       "            MAPPED_TERM_URI MAPPING_TYPE  \\\n",
       "0  EFO_0004264, EFO_0009431        Broad   \n",
       "1               EFO_0004616        Broad   \n",
       "2               EFO_0003778      ? Broad   \n",
       "3               EFO_0003901      ? Broad   \n",
       "4               EFO_0004247      ? Broad   \n",
       "\n",
       "  ICD10_CODE/SELF_REPORTED_TRAIT_FIELD_CODE COMMENTS/TICKET   AI  \n",
       "0                                       K55            DONE  NaN  \n",
       "1                                       M17            DONE  NaN  \n",
       "2                                       M07            DONE  NaN  \n",
       "3                                       R30            DONE  NaN  \n",
       "4                                       F38            DONE  NaN  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Broad', '? Broad', '? Exact', '? Narrow', 'Exact', 'Narrow', '?',\n",
       "       nan, 'Narrow?'], dtype=object)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map['MAPPING_TYPE'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(50, 7)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map[ukb_map['MAPPING_TYPE'] == 'Narrow'].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(572, 7)"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map[ukb_map['MAPPING_TYPE'] == 'Broad'].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(927, 7)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map[ukb_map['MAPPING_TYPE'] == 'Exact'].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ZOOMA QUERY</th>\n",
       "      <th>MAPPED_TERM_LABEL</th>\n",
       "      <th>MAPPED_TERM_URI</th>\n",
       "      <th>MAPPING_TYPE</th>\n",
       "      <th>ICD10_CODE/SELF_REPORTED_TRAIT_FIELD_CODE</th>\n",
       "      <th>COMMENTS/TICKET</th>\n",
       "      <th>AI</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>992</th>\n",
       "      <td>rodent ulcer</td>\n",
       "      <td>basal cell carcinoma</td>\n",
       "      <td>EFO_0004193</td>\n",
       "      <td>?</td>\n",
       "      <td>20001_1073</td>\n",
       "      <td>DONE</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      ZOOMA QUERY     MAPPED_TERM_LABEL MAPPED_TERM_URI MAPPING_TYPE  \\\n",
       "992  rodent ulcer  basal cell carcinoma     EFO_0004193            ?   \n",
       "\n",
       "    ICD10_CODE/SELF_REPORTED_TRAIT_FIELD_CODE COMMENTS/TICKET   AI  \n",
       "992                                20001_1073            DONE  NaN  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ukb_map[ukb_map['ICD10_CODE/SELF_REPORTED_TRAIT_FIELD_CODE'].str.contains('20001_1073')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Group traits according to EFO terms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "from efo import EFO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/miltondp/projects/labs/hakyimlab/phenomexcan/base/results/omim_silver_standard/data/EFO.csv.gz'"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "efo_file = os.path.join(conf.OMIM_SILVER_STANDARD_EFO_FILE)\n",
    "display(efo_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "efo = EFO(efo_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>class_id</th>\n",
       "      <th>omim_codes</th>\n",
       "      <th>preferred_label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_1390</td>\n",
       "      <td>UMLS:CN199356|MONDO:0015326|GARD:0003994|ICD10...</td>\n",
       "      <td>Night blindness - skeletal anomalies - dysmorp...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_1000880</td>\n",
       "      <td>SNOMEDCT:19161004|MSH:D016510|MONDO:0006713|IC...</td>\n",
       "      <td>corneal neovascularization</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0004381</td>\n",
       "      <td>NCIt:C48555</td>\n",
       "      <td>mole per liter</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>http://www.orpha.net/ORDO/Orphanet_69735</td>\n",
       "      <td>OMIM:607823|GARD:0012827|MONDO:0007670</td>\n",
       "      <td>Hypotrichosis - lymphedema - telangiectasia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>http://purl.obolibrary.org/obo/GO_0120062</td>\n",
       "      <td>NaN</td>\n",
       "      <td>positive regulation of gastric emptying</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_1000828</td>\n",
       "      <td>MSH:D015456|DOID:9953|MONDO:0006667|ICD10:C95.0</td>\n",
       "      <td>B- and T-cell mixed leukemia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>http://purl.obolibrary.org/obo/NCBITaxon_5076</td>\n",
       "      <td>SNOMEDCT:24406000|MSH:D010408</td>\n",
       "      <td>Penicillium chrysogenum</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0008234</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MHC class I polypeptide-related sequence B mea...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>http://purl.obolibrary.org/obo/CHEBI_53115</td>\n",
       "      <td>MSH:C081320</td>\n",
       "      <td>8-(3-chlorostyryl) caffeine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>http://purl.obolibrary.org/obo/CHEBI_17768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N-acetylputrescine</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        class_id  \\\n",
       "0        http://www.orpha.net/ORDO/Orphanet_1390   \n",
       "1           http://www.ebi.ac.uk/efo/EFO_1000880   \n",
       "2           http://www.ebi.ac.uk/efo/EFO_0004381   \n",
       "3       http://www.orpha.net/ORDO/Orphanet_69735   \n",
       "4      http://purl.obolibrary.org/obo/GO_0120062   \n",
       "5           http://www.ebi.ac.uk/efo/EFO_1000828   \n",
       "6  http://purl.obolibrary.org/obo/NCBITaxon_5076   \n",
       "7           http://www.ebi.ac.uk/efo/EFO_0008234   \n",
       "8     http://purl.obolibrary.org/obo/CHEBI_53115   \n",
       "9     http://purl.obolibrary.org/obo/CHEBI_17768   \n",
       "\n",
       "                                          omim_codes  \\\n",
       "0  UMLS:CN199356|MONDO:0015326|GARD:0003994|ICD10...   \n",
       "1  SNOMEDCT:19161004|MSH:D016510|MONDO:0006713|IC...   \n",
       "2                                        NCIt:C48555   \n",
       "3             OMIM:607823|GARD:0012827|MONDO:0007670   \n",
       "4                                                NaN   \n",
       "5    MSH:D015456|DOID:9953|MONDO:0006667|ICD10:C95.0   \n",
       "6                      SNOMEDCT:24406000|MSH:D010408   \n",
       "7                                                NaN   \n",
       "8                                        MSH:C081320   \n",
       "9                                                NaN   \n",
       "\n",
       "                                     preferred_label  \n",
       "0  Night blindness - skeletal anomalies - dysmorp...  \n",
       "1                         corneal neovascularization  \n",
       "2                                     mole per liter  \n",
       "3        Hypotrichosis - lymphedema - telangiectasia  \n",
       "4            positive regulation of gastric emptying  \n",
       "5                       B- and T-cell mixed leukemia  \n",
       "6                            Penicillium chrysogenum  \n",
       "7  MHC class I polypeptide-related sequence B mea...  \n",
       "8                        8-(3-chlorostyryl) caffeine  \n",
       "9                                 N-acetylputrescine  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "efo.efo_data_full.head(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ukb_data_with_efo = efo.assign_efo_from_ukb(ukb_data, map_types=('Exact',))\n",
    "ukb_data_with_efo = efo.assign_efo_from_ukb(ukb_data, map_types=('Exact', 'Broad', 'Narrow'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "ukb_data_with_efo = ukb_data_with_efo.dropna(subset=['efo_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1007, 4)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(ukb_data_with_efo.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ukb_fullcode</th>\n",
       "      <th>ukb_code</th>\n",
       "      <th>efo_code</th>\n",
       "      <th>efo_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>N49-Diagnoses_main_ICD10_N49_Inflammatory_diso...</td>\n",
       "      <td>N49</td>\n",
       "      <td>EFO_0009555</td>\n",
       "      <td>male reproductive system disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>S69-Diagnoses_main_ICD10_S69_Other_and_unspeci...</td>\n",
       "      <td>S69</td>\n",
       "      <td>EFO_0000546</td>\n",
       "      <td>injury</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>20002_1385-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1385</td>\n",
       "      <td>EFO_1001890</td>\n",
       "      <td>food allergy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>20002_1373-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1373</td>\n",
       "      <td>EFO_1001986</td>\n",
       "      <td>connective tissue disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>20002_1510-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1510</td>\n",
       "      <td>EFO_0008533</td>\n",
       "      <td>dyspepsia</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         ukb_fullcode    ukb_code  \\\n",
       "2   N49-Diagnoses_main_ICD10_N49_Inflammatory_diso...         N49   \n",
       "6   S69-Diagnoses_main_ICD10_S69_Other_and_unspeci...         S69   \n",
       "8   20002_1385-Noncancer_illness_code_selfreported...  20002_1385   \n",
       "13  20002_1373-Noncancer_illness_code_selfreported...  20002_1373   \n",
       "15  20002_1510-Noncancer_illness_code_selfreported...  20002_1510   \n",
       "\n",
       "       efo_code                          efo_name  \n",
       "2   EFO_0009555  male reproductive system disease  \n",
       "6   EFO_0000546                            injury  \n",
       "8   EFO_1001890                      food allergy  \n",
       "13  EFO_1001986         connective tissue disease  \n",
       "15  EFO_0008533                         dyspepsia  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "display(ukb_data_with_efo.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ukb_fullcode</th>\n",
       "      <th>ukb_code</th>\n",
       "      <th>efo_code</th>\n",
       "      <th>efo_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>991</th>\n",
       "      <td>J46-Diagnoses_main_ICD10_J46_Status_asthmaticus</td>\n",
       "      <td>J46</td>\n",
       "      <td>EFO_0008590</td>\n",
       "      <td>Status Asthmaticus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213</th>\n",
       "      <td>20002_1111-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1111</td>\n",
       "      <td>EFO_0000270</td>\n",
       "      <td>asthma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3485</th>\n",
       "      <td>22127-Doctor_diagnosed_asthma</td>\n",
       "      <td>22127</td>\n",
       "      <td>EFO_0000270</td>\n",
       "      <td>asthma</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3667</th>\n",
       "      <td>J45-Diagnoses_main_ICD10_J45_Asthma</td>\n",
       "      <td>J45</td>\n",
       "      <td>EFO_0000270</td>\n",
       "      <td>asthma</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           ukb_fullcode    ukb_code  \\\n",
       "991     J46-Diagnoses_main_ICD10_J46_Status_asthmaticus         J46   \n",
       "1213  20002_1111-Noncancer_illness_code_selfreported...  20002_1111   \n",
       "3485                      22127-Doctor_diagnosed_asthma       22127   \n",
       "3667                J45-Diagnoses_main_ICD10_J45_Asthma         J45   \n",
       "\n",
       "         efo_code            efo_name  \n",
       "991   EFO_0008590  Status Asthmaticus  \n",
       "1213  EFO_0000270              asthma  \n",
       "3485  EFO_0000270              asthma  \n",
       "3667  EFO_0000270              asthma  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# How does \"asthma\" get grouped?\n",
    "ukb_data_with_efo[ukb_data_with_efo['efo_name'].str.lower().str.contains('asthma')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save only ukb traits\n",
    "ukb_data_with_efo.to_csv('ukbiobank_efo_mappings.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Add GTEX GWAS traits description"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "import metadata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GTEx_GWAS</th>\n",
       "      <th>PUBMED_Paper_Link</th>\n",
       "      <th>Pheno_File</th>\n",
       "      <th>Source_File</th>\n",
       "      <th>Portal</th>\n",
       "      <th>Consortium</th>\n",
       "      <th>Link</th>\n",
       "      <th>Notes</th>\n",
       "      <th>Header</th>\n",
       "      <th>Hidden</th>\n",
       "      <th>...</th>\n",
       "      <th>Declared_Effect_Allele</th>\n",
       "      <th>Genome_Reference</th>\n",
       "      <th>Binary</th>\n",
       "      <th>Cases</th>\n",
       "      <th>abbreviation</th>\n",
       "      <th>new_abbreviation</th>\n",
       "      <th>new_Phenotype</th>\n",
       "      <th>Category</th>\n",
       "      <th>Deflation</th>\n",
       "      <th>color</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UKB_1160_Sleep_duration</th>\n",
       "      <td>Yes</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/showcase/field.cg...</td>\n",
       "      <td>1160.assoc.tsv.gz</td>\n",
       "      <td>1160.assoc.tsv.gz</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/</td>\n",
       "      <td>UK Biobank</td>\n",
       "      <td>https://www.dropbox.com/s/00v10oxwku3u5yq/1160...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>U_SL</td>\n",
       "      <td>SLEEP_UKB</td>\n",
       "      <td>Sleep_Duration_UKB</td>\n",
       "      <td>Psychiatric-neurologic</td>\n",
       "      <td>0</td>\n",
       "      <td>#0000A0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UKB_1180_Morning_or_evening_person_chronotype</th>\n",
       "      <td>Yes</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/showcase/field.cg...</td>\n",
       "      <td>1180.assoc.tsv.gz</td>\n",
       "      <td>1180.assoc.tsv.gz</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/</td>\n",
       "      <td>UK Biobank</td>\n",
       "      <td>https://www.dropbox.com/s/z7cppp7ndmylpf0/1180...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>U_CHR</td>\n",
       "      <td>CHRONO_UKB</td>\n",
       "      <td>Chronotype_UKB</td>\n",
       "      <td>Psychiatric-neurologic</td>\n",
       "      <td>0</td>\n",
       "      <td>#0000A0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UKB_1200_Sleeplessness_or_insomnia</th>\n",
       "      <td>Yes</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/showcase/field.cg...</td>\n",
       "      <td>1200.assoc.tsv.gz</td>\n",
       "      <td>1200.assoc.tsv.gz</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/</td>\n",
       "      <td>UK Biobank</td>\n",
       "      <td>https://www.dropbox.com/s/ugt1g44ycju9ms5/1200...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>U_NSL</td>\n",
       "      <td>INSOMN_UKB</td>\n",
       "      <td>Insomnia_UKB</td>\n",
       "      <td>Psychiatric-neurologic</td>\n",
       "      <td>0</td>\n",
       "      <td>#0000A0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UKB_1807_Fathers_age_at_death</th>\n",
       "      <td>Yes</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/showcase/field.cg...</td>\n",
       "      <td>1807.assoc.tsv.gz</td>\n",
       "      <td>1807.assoc.tsv.gz</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/</td>\n",
       "      <td>UK Biobank</td>\n",
       "      <td>https://www.dropbox.com/s/qftvyi5a9qkb5xw/1807...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>U_FAD</td>\n",
       "      <td>FAD_UKB</td>\n",
       "      <td>Fathers_Age_At_Death_UKB</td>\n",
       "      <td>Aging</td>\n",
       "      <td>0</td>\n",
       "      <td>#400040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UKB_20002_1094_self_reported_deep_venous_thrombosis_dvt</th>\n",
       "      <td>Yes</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/showcase/field.cg...</td>\n",
       "      <td>20002_1094.assoc.tsv.gz</td>\n",
       "      <td>20002_1094.assoc.tsv.gz</td>\n",
       "      <td>http://biobank.ctsu.ox.ac.uk/</td>\n",
       "      <td>UK Biobank</td>\n",
       "      <td>https://www.dropbox.com/s/0smh60npg2eihqv/2000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>6767.0</td>\n",
       "      <td>UN_DVT</td>\n",
       "      <td>DVT_UKBS</td>\n",
       "      <td>Deep_Venous_Thrombosis_UKBS</td>\n",
       "      <td>Cardiometabolic</td>\n",
       "      <td>0</td>\n",
       "      <td>#004000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   GTEx_GWAS  \\\n",
       "Tag                                                            \n",
       "UKB_1160_Sleep_duration                                  Yes   \n",
       "UKB_1180_Morning_or_evening_person_chronotype            Yes   \n",
       "UKB_1200_Sleeplessness_or_insomnia                       Yes   \n",
       "UKB_1807_Fathers_age_at_death                            Yes   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...       Yes   \n",
       "\n",
       "                                                                                    PUBMED_Paper_Link  \\\n",
       "Tag                                                                                                     \n",
       "UKB_1160_Sleep_duration                             http://biobank.ctsu.ox.ac.uk/showcase/field.cg...   \n",
       "UKB_1180_Morning_or_evening_person_chronotype       http://biobank.ctsu.ox.ac.uk/showcase/field.cg...   \n",
       "UKB_1200_Sleeplessness_or_insomnia                  http://biobank.ctsu.ox.ac.uk/showcase/field.cg...   \n",
       "UKB_1807_Fathers_age_at_death                       http://biobank.ctsu.ox.ac.uk/showcase/field.cg...   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  http://biobank.ctsu.ox.ac.uk/showcase/field.cg...   \n",
       "\n",
       "                                                                 Pheno_File  \\\n",
       "Tag                                                                           \n",
       "UKB_1160_Sleep_duration                                   1160.assoc.tsv.gz   \n",
       "UKB_1180_Morning_or_evening_person_chronotype             1180.assoc.tsv.gz   \n",
       "UKB_1200_Sleeplessness_or_insomnia                        1200.assoc.tsv.gz   \n",
       "UKB_1807_Fathers_age_at_death                             1807.assoc.tsv.gz   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  20002_1094.assoc.tsv.gz   \n",
       "\n",
       "                                                                Source_File  \\\n",
       "Tag                                                                           \n",
       "UKB_1160_Sleep_duration                                   1160.assoc.tsv.gz   \n",
       "UKB_1180_Morning_or_evening_person_chronotype             1180.assoc.tsv.gz   \n",
       "UKB_1200_Sleeplessness_or_insomnia                        1200.assoc.tsv.gz   \n",
       "UKB_1807_Fathers_age_at_death                             1807.assoc.tsv.gz   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  20002_1094.assoc.tsv.gz   \n",
       "\n",
       "                                                                           Portal  \\\n",
       "Tag                                                                                 \n",
       "UKB_1160_Sleep_duration                             http://biobank.ctsu.ox.ac.uk/   \n",
       "UKB_1180_Morning_or_evening_person_chronotype       http://biobank.ctsu.ox.ac.uk/   \n",
       "UKB_1200_Sleeplessness_or_insomnia                  http://biobank.ctsu.ox.ac.uk/   \n",
       "UKB_1807_Fathers_age_at_death                       http://biobank.ctsu.ox.ac.uk/   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  http://biobank.ctsu.ox.ac.uk/   \n",
       "\n",
       "                                                    Consortium  \\\n",
       "Tag                                                              \n",
       "UKB_1160_Sleep_duration                             UK Biobank   \n",
       "UKB_1180_Morning_or_evening_person_chronotype       UK Biobank   \n",
       "UKB_1200_Sleeplessness_or_insomnia                  UK Biobank   \n",
       "UKB_1807_Fathers_age_at_death                       UK Biobank   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  UK Biobank   \n",
       "\n",
       "                                                                                                 Link  \\\n",
       "Tag                                                                                                     \n",
       "UKB_1160_Sleep_duration                             https://www.dropbox.com/s/00v10oxwku3u5yq/1160...   \n",
       "UKB_1180_Morning_or_evening_person_chronotype       https://www.dropbox.com/s/z7cppp7ndmylpf0/1180...   \n",
       "UKB_1200_Sleeplessness_or_insomnia                  https://www.dropbox.com/s/ugt1g44ycju9ms5/1200...   \n",
       "UKB_1807_Fathers_age_at_death                       https://www.dropbox.com/s/qftvyi5a9qkb5xw/1807...   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  https://www.dropbox.com/s/0smh60npg2eihqv/2000...   \n",
       "\n",
       "                                                   Notes Header Hidden  ...  \\\n",
       "Tag                                                                     ...   \n",
       "UKB_1160_Sleep_duration                              NaN    NaN    NaN  ...   \n",
       "UKB_1180_Morning_or_evening_person_chronotype        NaN    NaN    NaN  ...   \n",
       "UKB_1200_Sleeplessness_or_insomnia                   NaN    NaN    NaN  ...   \n",
       "UKB_1807_Fathers_age_at_death                        NaN    NaN    NaN  ...   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...   NaN    NaN    NaN  ...   \n",
       "\n",
       "                                                   Declared_Effect_Allele  \\\n",
       "Tag                                                                         \n",
       "UKB_1160_Sleep_duration                                               NaN   \n",
       "UKB_1180_Morning_or_evening_person_chronotype                         NaN   \n",
       "UKB_1200_Sleeplessness_or_insomnia                                    NaN   \n",
       "UKB_1807_Fathers_age_at_death                                         NaN   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...                    NaN   \n",
       "\n",
       "                                                   Genome_Reference Binary  \\\n",
       "Tag                                                                          \n",
       "UKB_1160_Sleep_duration                                         NaN      0   \n",
       "UKB_1180_Morning_or_evening_person_chronotype                   NaN      0   \n",
       "UKB_1200_Sleeplessness_or_insomnia                              NaN      0   \n",
       "UKB_1807_Fathers_age_at_death                                   NaN      0   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...              NaN      1   \n",
       "\n",
       "                                                     Cases abbreviation  \\\n",
       "Tag                                                                       \n",
       "UKB_1160_Sleep_duration                                NaN         U_SL   \n",
       "UKB_1180_Morning_or_evening_person_chronotype          NaN        U_CHR   \n",
       "UKB_1200_Sleeplessness_or_insomnia                     NaN        U_NSL   \n",
       "UKB_1807_Fathers_age_at_death                          NaN        U_FAD   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  6767.0       UN_DVT   \n",
       "\n",
       "                                                    new_abbreviation  \\\n",
       "Tag                                                                    \n",
       "UKB_1160_Sleep_duration                                    SLEEP_UKB   \n",
       "UKB_1180_Morning_or_evening_person_chronotype             CHRONO_UKB   \n",
       "UKB_1200_Sleeplessness_or_insomnia                        INSOMN_UKB   \n",
       "UKB_1807_Fathers_age_at_death                                FAD_UKB   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...          DVT_UKBS   \n",
       "\n",
       "                                                                  new_Phenotype  \\\n",
       "Tag                                                                               \n",
       "UKB_1160_Sleep_duration                                      Sleep_Duration_UKB   \n",
       "UKB_1180_Morning_or_evening_person_chronotype                    Chronotype_UKB   \n",
       "UKB_1200_Sleeplessness_or_insomnia                                 Insomnia_UKB   \n",
       "UKB_1807_Fathers_age_at_death                          Fathers_Age_At_Death_UKB   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...  Deep_Venous_Thrombosis_UKBS   \n",
       "\n",
       "                                                                  Category  \\\n",
       "Tag                                                                          \n",
       "UKB_1160_Sleep_duration                             Psychiatric-neurologic   \n",
       "UKB_1180_Morning_or_evening_person_chronotype       Psychiatric-neurologic   \n",
       "UKB_1200_Sleeplessness_or_insomnia                  Psychiatric-neurologic   \n",
       "UKB_1807_Fathers_age_at_death                                        Aging   \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...         Cardiometabolic   \n",
       "\n",
       "                                                   Deflation    color  \n",
       "Tag                                                                    \n",
       "UKB_1160_Sleep_duration                                    0  #0000A0  \n",
       "UKB_1180_Morning_or_evening_person_chronotype              0  #0000A0  \n",
       "UKB_1200_Sleeplessness_or_insomnia                         0  #0000A0  \n",
       "UKB_1807_Fathers_age_at_death                              0  #400040  \n",
       "UKB_20002_1094_self_reported_deep_venous_thromb...         0  #004000  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.GTEX_GWAS_PHENO_INFO.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_gwas_traits_info = metadata.GTEX_GWAS_PHENO_INFO[['EFO', 'HPO', 'Description', 'Phenotype']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_gwas_traits_info = gtex_gwas_traits_info[gtex_gwas_traits_info.index.isin(smultixcan_genes_associations.columns)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(42, 4)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_gwas_traits_info.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EFO</th>\n",
       "      <th>HPO</th>\n",
       "      <th>Description</th>\n",
       "      <th>Phenotype</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Eosinophil_counts</th>\n",
       "      <td>EFO_0004842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Eosinophil counts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Granulocyte_count</th>\n",
       "      <td>EFO_0007987</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Granulocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_High_light_scatter_reticulocyte_count</th>\n",
       "      <td>EFO_0007986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>High light scatter reticulocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Lymphocyte_counts</th>\n",
       "      <td>EFO_0004587</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lymphocyte counts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Monocyte_count</th>\n",
       "      <td>EFO_0005091</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Monocyte count</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            EFO  HPO  \\\n",
       "Tag                                                                    \n",
       "Astle_et_al_2016_Eosinophil_counts                  EFO_0004842  NaN   \n",
       "Astle_et_al_2016_Granulocyte_count                  EFO_0007987  NaN   \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...  EFO_0007986  NaN   \n",
       "Astle_et_al_2016_Lymphocyte_counts                  EFO_0004587  NaN   \n",
       "Astle_et_al_2016_Monocyte_count                     EFO_0005091  NaN   \n",
       "\n",
       "                                                    Description  \\\n",
       "Tag                                                               \n",
       "Astle_et_al_2016_Eosinophil_counts                          NaN   \n",
       "Astle_et_al_2016_Granulocyte_count                          NaN   \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...          NaN   \n",
       "Astle_et_al_2016_Lymphocyte_counts                          NaN   \n",
       "Astle_et_al_2016_Monocyte_count                             NaN   \n",
       "\n",
       "                                                                                Phenotype  \n",
       "Tag                                                                                        \n",
       "Astle_et_al_2016_Eosinophil_counts                                      Eosinophil counts  \n",
       "Astle_et_al_2016_Granulocyte_count                                      Granulocyte count  \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...  High light scatter reticulocyte count  \n",
       "Astle_et_al_2016_Lymphocyte_counts                                      Lymphocyte counts  \n",
       "Astle_et_al_2016_Monocyte_count                                            Monocyte count  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_gwas_traits_info.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_gwas_data_with_efo = efo.assign_efo_label_from_efo_code(gtex_gwas_traits_info, efo_code_column='EFO')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>EFO</th>\n",
       "      <th>HPO</th>\n",
       "      <th>Description</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>efo_name</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tag</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Eosinophil_counts</th>\n",
       "      <td>EFO_0004842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Eosinophil counts</td>\n",
       "      <td>eosinophil count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Granulocyte_count</th>\n",
       "      <td>EFO_0007987</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Granulocyte count</td>\n",
       "      <td>granulocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_High_light_scatter_reticulocyte_count</th>\n",
       "      <td>EFO_0007986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>High light scatter reticulocyte count</td>\n",
       "      <td>reticulocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Lymphocyte_counts</th>\n",
       "      <td>EFO_0004587</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lymphocyte counts</td>\n",
       "      <td>lymphocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Astle_et_al_2016_Monocyte_count</th>\n",
       "      <td>EFO_0005091</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Monocyte count</td>\n",
       "      <td>monocyte count</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                            EFO  HPO  \\\n",
       "Tag                                                                    \n",
       "Astle_et_al_2016_Eosinophil_counts                  EFO_0004842  NaN   \n",
       "Astle_et_al_2016_Granulocyte_count                  EFO_0007987  NaN   \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...  EFO_0007986  NaN   \n",
       "Astle_et_al_2016_Lymphocyte_counts                  EFO_0004587  NaN   \n",
       "Astle_et_al_2016_Monocyte_count                     EFO_0005091  NaN   \n",
       "\n",
       "                                                    Description  \\\n",
       "Tag                                                               \n",
       "Astle_et_al_2016_Eosinophil_counts                          NaN   \n",
       "Astle_et_al_2016_Granulocyte_count                          NaN   \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...          NaN   \n",
       "Astle_et_al_2016_Lymphocyte_counts                          NaN   \n",
       "Astle_et_al_2016_Monocyte_count                             NaN   \n",
       "\n",
       "                                                                                Phenotype  \\\n",
       "Tag                                                                                         \n",
       "Astle_et_al_2016_Eosinophil_counts                                      Eosinophil counts   \n",
       "Astle_et_al_2016_Granulocyte_count                                      Granulocyte count   \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...  High light scatter reticulocyte count   \n",
       "Astle_et_al_2016_Lymphocyte_counts                                      Lymphocyte counts   \n",
       "Astle_et_al_2016_Monocyte_count                                            Monocyte count   \n",
       "\n",
       "                                                              efo_name  \n",
       "Tag                                                                     \n",
       "Astle_et_al_2016_Eosinophil_counts                    eosinophil count  \n",
       "Astle_et_al_2016_Granulocyte_count                   granulocyte count  \n",
       "Astle_et_al_2016_High_light_scatter_reticulocyt...  reticulocyte count  \n",
       "Astle_et_al_2016_Lymphocyte_counts                    lymphocyte count  \n",
       "Astle_et_al_2016_Monocyte_count                         monocyte count  "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_gwas_data_with_efo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_gwas_data_with_efo = gtex_gwas_data_with_efo[['efo_name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_gwas_data_with_efo = gtex_gwas_data_with_efo.assign(ukb_code=gtex_gwas_data_with_efo.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "gtex_gwas_data_with_efo = gtex_gwas_data_with_efo.reset_index()[['Tag', 'ukb_code', 'efo_name']].rename(columns={'Tag': 'ukb_fullcode'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ukb_fullcode</th>\n",
       "      <th>ukb_code</th>\n",
       "      <th>efo_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Astle_et_al_2016_Eosinophil_counts</td>\n",
       "      <td>Astle_et_al_2016_Eosinophil_counts</td>\n",
       "      <td>eosinophil count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Astle_et_al_2016_Granulocyte_count</td>\n",
       "      <td>Astle_et_al_2016_Granulocyte_count</td>\n",
       "      <td>granulocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Astle_et_al_2016_High_light_scatter_reticulocy...</td>\n",
       "      <td>Astle_et_al_2016_High_light_scatter_reticulocy...</td>\n",
       "      <td>reticulocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Astle_et_al_2016_Lymphocyte_counts</td>\n",
       "      <td>Astle_et_al_2016_Lymphocyte_counts</td>\n",
       "      <td>lymphocyte count</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Astle_et_al_2016_Monocyte_count</td>\n",
       "      <td>Astle_et_al_2016_Monocyte_count</td>\n",
       "      <td>monocyte count</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        ukb_fullcode  \\\n",
       "0                 Astle_et_al_2016_Eosinophil_counts   \n",
       "1                 Astle_et_al_2016_Granulocyte_count   \n",
       "2  Astle_et_al_2016_High_light_scatter_reticulocy...   \n",
       "3                 Astle_et_al_2016_Lymphocyte_counts   \n",
       "4                    Astle_et_al_2016_Monocyte_count   \n",
       "\n",
       "                                            ukb_code            efo_name  \n",
       "0                 Astle_et_al_2016_Eosinophil_counts    eosinophil count  \n",
       "1                 Astle_et_al_2016_Granulocyte_count   granulocyte count  \n",
       "2  Astle_et_al_2016_High_light_scatter_reticulocy...  reticulocyte count  \n",
       "3                 Astle_et_al_2016_Lymphocyte_counts    lymphocyte count  \n",
       "4                    Astle_et_al_2016_Monocyte_count      monocyte count  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gtex_gwas_data_with_efo.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.concat((ukb_data_with_efo, gtex_gwas_data_with_efo), ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# remove efo_name with nan\n",
    "df = df.dropna(subset=['efo_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(643, 3)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ukb_fullcode</th>\n",
       "      <th>ukb_code</th>\n",
       "      <th>efo_name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20002_1385-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1385</td>\n",
       "      <td>food allergy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20002_1373-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1373</td>\n",
       "      <td>connective tissue disease</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20002_1510-Noncancer_illness_code_selfreported...</td>\n",
       "      <td>20002_1510</td>\n",
       "      <td>dyspepsia</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O48-Diagnoses_main_ICD10_O48_Prolonged_pregnancy</td>\n",
       "      <td>O48</td>\n",
       "      <td>post term pregnancy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>N17-Diagnoses_main_ICD10_N17_Acute_renal_failure</td>\n",
       "      <td>N17</td>\n",
       "      <td>Acute kidney injury</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        ukb_fullcode    ukb_code  \\\n",
       "0  20002_1385-Noncancer_illness_code_selfreported...  20002_1385   \n",
       "1  20002_1373-Noncancer_illness_code_selfreported...  20002_1373   \n",
       "2  20002_1510-Noncancer_illness_code_selfreported...  20002_1510   \n",
       "3   O48-Diagnoses_main_ICD10_O48_Prolonged_pregnancy         O48   \n",
       "4   N17-Diagnoses_main_ICD10_N17_Acute_renal_failure         N17   \n",
       "\n",
       "                    efo_name  \n",
       "0               food allergy  \n",
       "1  connective tissue disease  \n",
       "2                  dyspepsia  \n",
       "3        post term pregnancy  \n",
       "4        Acute kidney injury  "
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.to_csv('ukbiobank_efo_mappings.tsv', index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run map_trait_to_hpo_and_mim.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "omim_silver_standard_dir = conf.OMIM_SILVER_STANDARD_DATA_DIR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: â€˜dplyrâ€™\n",
      "\n",
      "The following objects are masked from â€˜package:statsâ€™:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from â€˜package:baseâ€™:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$omim_silver_standard_dir\"\n",
    "Rscript map_trait_to_hpo_and_mim.R $1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run mim-to-gene.R (generates OMIM silver standard)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Attaching package: â€˜dplyrâ€™\n",
      "\n",
      "The following objects are masked from â€˜package:data.tableâ€™:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "The following objects are masked from â€˜package:statsâ€™:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from â€˜package:baseâ€™:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n",
      "Taking input= as a system command ('cat /home/miltondp/projects/labs/hakyimlab/phenomexcan/base/results/omim_silver_standard/data/mim2gene.txt | tail -n +5') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure envionment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.\n",
      "Taking input= as a system command ('cat /home/miltondp/projects/labs/hakyimlab/phenomexcan/base/results/omim_silver_standard/data/genemap2.txt | head -n 16943 | tail -n +4') and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure envionment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.\n",
      "1\n",
      "2\n",
      "3\n",
      "4\n",
      "5\n",
      "6\n",
      "7\n",
      "8\n",
      "9\n",
      "10\n",
      "11\n",
      "12\n",
      "13\n",
      "14\n",
      "15\n",
      "16\n",
      "17\n",
      "18\n",
      "19\n",
      "20\n",
      "21\n",
      "22\n",
      "23\n",
      "24\n",
      "25\n",
      "26\n",
      "27\n",
      "28\n",
      "29\n",
      "30\n",
      "31\n",
      "32\n",
      "33\n",
      "34\n",
      "35\n",
      "36\n",
      "37\n",
      "38\n",
      "39\n",
      "40\n",
      "41\n",
      "42\n",
      "43\n",
      "44\n",
      "45\n",
      "46\n",
      "47\n",
      "48\n",
      "49\n",
      "50\n",
      "51\n",
      "52\n",
      "53\n",
      "54\n",
      "55\n",
      "56\n",
      "57\n",
      "58\n",
      "59\n",
      "60\n",
      "61\n",
      "62\n",
      "63\n",
      "64\n",
      "65\n",
      "66\n",
      "67\n",
      "68\n",
      "69\n",
      "70\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "75\n",
      "76\n",
      "77\n",
      "78\n",
      "79\n",
      "80\n",
      "81\n",
      "82\n",
      "83\n",
      "84\n",
      "85\n",
      "86\n",
      "87\n",
      "88\n",
      "89\n",
      "90\n",
      "91\n",
      "92\n",
      "93\n",
      "94\n",
      "95\n",
      "96\n",
      "97\n",
      "98\n",
      "99\n",
      "100\n",
      "101\n",
      "102\n",
      "103\n",
      "104\n",
      "105\n",
      "106\n",
      "107\n",
      "108\n",
      "109\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "116\n",
      "117\n",
      "118\n",
      "119\n",
      "120\n",
      "121\n",
      "122\n",
      "123\n",
      "124\n",
      "125\n",
      "126\n",
      "127\n",
      "128\n",
      "129\n",
      "130\n",
      "131\n",
      "132\n",
      "133\n",
      "134\n",
      "135\n",
      "136\n",
      "137\n",
      "138\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "143\n",
      "144\n",
      "145\n",
      "146\n",
      "147\n",
      "148\n",
      "149\n",
      "150\n",
      "151\n",
      "152\n",
      "153\n",
      "154\n",
      "155\n",
      "156\n",
      "157\n",
      "158\n",
      "159\n",
      "160\n",
      "161\n",
      "162\n",
      "163\n",
      "164\n",
      "165\n",
      "166\n",
      "167\n",
      "168\n",
      "169\n",
      "170\n",
      "171\n",
      "172\n",
      "173\n",
      "174\n",
      "175\n",
      "176\n",
      "177\n",
      "178\n",
      "179\n",
      "180\n",
      "181\n",
      "182\n",
      "183\n",
      "184\n",
      "185\n",
      "186\n",
      "187\n",
      "188\n",
      "189\n",
      "190\n",
      "191\n",
      "192\n",
      "193\n",
      "194\n",
      "195\n",
      "196\n",
      "197\n",
      "198\n",
      "199\n",
      "200\n",
      "201\n",
      "202\n",
      "203\n",
      "204\n",
      "205\n",
      "206\n",
      "207\n",
      "208\n",
      "209\n",
      "210\n",
      "211\n",
      "212\n",
      "213\n",
      "214\n",
      "215\n",
      "216\n",
      "217\n",
      "218\n",
      "219\n",
      "220\n",
      "221\n",
      "222\n",
      "223\n",
      "224\n",
      "225\n",
      "226\n",
      "227\n",
      "228\n",
      "229\n",
      "230\n",
      "231\n",
      "232\n",
      "233\n",
      "234\n",
      "235\n",
      "236\n",
      "237\n",
      "238\n",
      "239\n",
      "240\n",
      "241\n",
      "242\n",
      "243\n",
      "244\n",
      "245\n",
      "246\n",
      "247\n",
      "248\n",
      "249\n",
      "250\n",
      "251\n",
      "252\n",
      "253\n",
      "254\n",
      "255\n",
      "256\n",
      "257\n",
      "258\n",
      "259\n",
      "260\n",
      "261\n",
      "262\n",
      "263\n",
      "264\n",
      "265\n",
      "266\n",
      "267\n",
      "268\n",
      "269\n",
      "270\n",
      "271\n",
      "272\n",
      "273\n",
      "274\n",
      "275\n",
      "276\n",
      "277\n",
      "278\n",
      "279\n",
      "280\n",
      "281\n",
      "282\n",
      "283\n",
      "284\n",
      "285\n",
      "286\n",
      "287\n",
      "288\n",
      "289\n",
      "290\n",
      "291\n",
      "292\n",
      "293\n",
      "294\n",
      "295\n",
      "296\n",
      "297\n",
      "298\n",
      "299\n",
      "300\n",
      "301\n",
      "302\n",
      "303\n",
      "304\n",
      "305\n",
      "306\n",
      "307\n",
      "308\n",
      "309\n",
      "310\n",
      "311\n",
      "312\n",
      "313\n",
      "314\n",
      "315\n",
      "316\n",
      "317\n",
      "318\n",
      "319\n",
      "320\n",
      "321\n",
      "322\n",
      "323\n",
      "324\n",
      "325\n",
      "326\n",
      "327\n",
      "328\n",
      "329\n",
      "330\n",
      "331\n",
      "332\n",
      "333\n",
      "334\n",
      "335\n",
      "336\n",
      "337\n",
      "338\n",
      "339\n",
      "340\n",
      "341\n",
      "342\n",
      "343\n",
      "344\n",
      "345\n",
      "346\n",
      "347\n",
      "348\n",
      "349\n",
      "350\n",
      "351\n",
      "352\n",
      "353\n",
      "354\n",
      "355\n",
      "356\n",
      "357\n",
      "358\n",
      "359\n",
      "360\n",
      "361\n",
      "362\n",
      "363\n",
      "364\n",
      "365\n",
      "366\n",
      "367\n",
      "368\n",
      "369\n",
      "370\n",
      "371\n",
      "372\n",
      "373\n",
      "374\n",
      "375\n",
      "376\n",
      "377\n",
      "378\n",
      "379\n",
      "380\n",
      "381\n",
      "382\n",
      "383\n",
      "384\n",
      "385\n",
      "386\n",
      "387\n",
      "388\n",
      "389\n",
      "390\n",
      "391\n",
      "392\n",
      "393\n",
      "394\n",
      "395\n",
      "396\n",
      "397\n",
      "398\n",
      "399\n",
      "400\n",
      "401\n",
      "402\n",
      "403\n",
      "404\n",
      "405\n",
      "406\n",
      "407\n",
      "408\n",
      "409\n",
      "410\n",
      "411\n",
      "412\n",
      "413\n",
      "414\n",
      "415\n",
      "416\n",
      "417\n",
      "418\n",
      "419\n",
      "420\n",
      "421\n",
      "422\n",
      "423\n",
      "424\n",
      "425\n",
      "426\n",
      "427\n",
      "428\n",
      "429\n",
      "430\n",
      "431\n",
      "432\n",
      "433\n",
      "434\n",
      "435\n",
      "436\n",
      "437\n",
      "438\n",
      "439\n",
      "440\n",
      "441\n",
      "442\n",
      "443\n",
      "444\n",
      "445\n",
      "446\n",
      "447\n",
      "448\n",
      "449\n",
      "450\n",
      "451\n",
      "452\n",
      "453\n",
      "454\n",
      "455\n",
      "456\n",
      "457\n",
      "458\n",
      "459\n",
      "460\n",
      "461\n",
      "462\n",
      "463\n",
      "464\n",
      "465\n",
      "466\n",
      "467\n",
      "468\n",
      "469\n",
      "470\n",
      "471\n",
      "472\n",
      "473\n",
      "474\n",
      "475\n",
      "476\n",
      "477\n",
      "478\n",
      "479\n",
      "480\n",
      "481\n",
      "482\n",
      "483\n",
      "484\n",
      "485\n",
      "486\n",
      "487\n",
      "488\n",
      "489\n",
      "490\n",
      "491\n",
      "492\n",
      "493\n",
      "494\n",
      "495\n",
      "496\n",
      "497\n",
      "498\n",
      "499\n",
      "500\n",
      "501\n",
      "502\n",
      "503\n",
      "504\n",
      "505\n",
      "506\n",
      "507\n",
      "508\n",
      "509\n",
      "510\n",
      "511\n",
      "512\n",
      "513\n",
      "514\n",
      "515\n",
      "516\n",
      "517\n",
      "518\n",
      "519\n",
      "520\n",
      "521\n",
      "522\n",
      "523\n",
      "524\n",
      "525\n",
      "526\n",
      "527\n",
      "528\n",
      "529\n",
      "530\n",
      "531\n",
      "532\n",
      "533\n",
      "534\n",
      "535\n",
      "536\n",
      "537\n",
      "538\n",
      "539\n",
      "540\n",
      "541\n",
      "542\n",
      "543\n",
      "544\n",
      "545\n",
      "546\n",
      "547\n",
      "548\n",
      "549\n",
      "550\n",
      "551\n",
      "552\n",
      "553\n",
      "554\n",
      "555\n",
      "556\n",
      "557\n",
      "558\n",
      "559\n",
      "560\n",
      "561\n",
      "562\n",
      "563\n",
      "564\n",
      "565\n",
      "566\n",
      "567\n",
      "568\n",
      "569\n",
      "570\n",
      "571\n",
      "572\n",
      "573\n",
      "574\n",
      "575\n",
      "576\n",
      "577\n",
      "578\n",
      "579\n",
      "580\n",
      "581\n",
      "582\n",
      "583\n",
      "584\n",
      "585\n",
      "586\n",
      "587\n",
      "588\n",
      "589\n",
      "590\n",
      "591\n",
      "592\n",
      "593\n",
      "594\n",
      "595\n",
      "596\n",
      "597\n",
      "598\n",
      "599\n",
      "600\n",
      "601\n",
      "602\n",
      "603\n",
      "604\n",
      "605\n",
      "606\n",
      "607\n",
      "608\n",
      "609\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"$omim_silver_standard_dir\"\n",
    "Rscript mim-to-gene.R $1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Move OMIM silver standard file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_omim_file_location = conf.OMIM_SILVER_STANDARD_FINAL_FILE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/miltondp/projects/labs/hakyimlab/phenomexcan/base/results/omim_silver_standard/data/omim_silver_standard.tsv'"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_omim_file_location"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash -s \"$new_omim_file_location\"\n",
    "mv omim_silver_standard.tsv $1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
